theano/tensor/opt.py

```diff
@@ -7490,23 +7490,20 @@ def local_useless_composite(node):
 # Although the ops ConsiderConstant, ZeroGrad and DisconnectedGrad
 # just returns the input, it should be removed from the graph to
 # make sure all possible optimizations can be applied.
-register_canonicalize(gof.OpRemove(theano.gradient.consider_constant_),
-                      'fast_compile', 'fast_run',
-                      name='remove_consider_constant')
-
-register_canonicalize(gof.OpRemove(theano.gradient.zero_grad_),
-                      'fast_compile', 'fast_run', name='remove_zero_grad')
-
-register_canonicalize(gof.OpRemove(theano.gradient.disconnected_grad_),
-                      'fast_compile', 'fast_run',
-                      name='remove_disconnected_grad')
-
-
-@register_canonicalize
-@gof.local_optimizer([theano.gradient.GradClip])
-def local_grad_clip(node):
-    if isinstance(node.op, theano.gradient.GradClip):
-        return node.inputs
+for o, name in [(gof.OpRemove(theano.gradient.consider_constant_),
+                 "remove_consider_constant"),
+                (gof.OpRemove(theano.gradient.zero_grad_),
+                 'remove_zero_grad'),
+                (gof.OpRemove(theano.gradient.disconnected_grad_),
+                 'remove_disconnected_grad'),
+                (gof.OpRemove(theano.gradient.undefined_grad_),
+                 'remove_undefined_grad'),
+                (gof.OpRemove(theano.gradient.GradClip),
+                 'remove_grad_clip'),
+                (gof.OpRemove(theano.gradient.GradScale),
+                 'remove_grad_scale')]:
+    register_canonicalize(o, 'fast_compile', 'fast_run', name=name)
+    register_useless(o, 'fast_compile', 'fast_run', name=name)
```

Comment:
Isn't the point of this to avoid this listing and just have one opt working on ViewOp?